lib/EnsEMBL/REST/Controller/VariantRecoder.pm

```diff
@@ -0,0 +1,147 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute 
+Copyright [2016-2017] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package EnsEMBL::REST::Controller::VariantRecoder;
+use Moose;
+use namespace::autoclean;
+use Data::Dumper;
+use Bio::EnsEMBL::VEP::VariantRecoder;
+
+use Try::Tiny;
+
+require EnsEMBL::REST;
+
+EnsEMBL::REST->turn_on_config_serialisers(__PACKAGE__);
+BEGIN { 
+  extends 'Catalyst::Controller::REST';
+}
+
+with 'EnsEMBL::REST::Role::PostLimiter';
+
+# /variant_recoder/:species
+sub species : Chained('/') PathPart('variant_recoder') CaptureArgs(1) {
+  my ( $self, $c, $species ) = @_;
+  my $reg = $c->model('Registry');
+  $c->stash->{species} = $reg->get_alias($species);
+
+  my $csa = $reg->get_adaptor($species, 'core', 'coordsystem');
+  $c->go('ReturnError', 'from_ensembl', ['Unable to fetch CoordSystem adaptor']) unless $csa;
+  my $css = $csa->fetch_all;
+  $c->go('ReturnError', 'from_ensembl', ['No coordinate systems found']) unless $css && @$css;
+
+  $c->stash->{assembly} = $css->[0]->version;
+  $c->stash->{has_variation} = $c->model('Registry')->get_adaptor( $species, 'Variation', 'Variation');
+  $c->log->debug('Working with species '.$species);
+}
+
+
+# /variant_recoder/:species/:id
+sub id: Chained('species') PathPart('') ActionClass('REST') {}
+
+sub id_GET {
+  my ( $self, $c, $rs_id ) = @_;
+
+  if (!$c->stash->{has_variation}) { $c->go('ReturnError', 'custom', ["Species ".$c->stash->{species}." does not have a variation database"]); }
+  
+  unless ($rs_id) {$c->go('ReturnError', 'custom', ["variant ID is a required parameter for this endpoint"])}
+
+  my $user_config = $c->request->parameters;
+  my $config = $self->_include_user_params($c,$user_config);
+
+  $self->get_results($c, $config, $rs_id);
+}
+
+sub id_POST {
+  my ($self, $c) = @_;
+
+  if (!$c->stash->{has_variation}) { $c->go('ReturnError', 'custom', ["Species ".$c->stash->{species}." does not have a variation database"]); }
```

Comment:
this can probably go in the sub id, as it is shared by both the GET and POST methods